### PR TITLE
LazyTSDBOpener to unblock HTTP handler for metrics and profile when replaying WAL

### LINKED ISSUE #37: Start http listener before opening the DB
I restarted `conprof` and it has been silent for 25 minutes; if it was listening on `/debug/pprof` I could find out what it is doing.  Also it should respond to `/metrics`.

### PR DESCRIPTION
**Pull request by metalmatze**
Closes #37 
Other components might need updating as well.

cc @bboreham

### CODE REVIEW

**Comment by metalmatze** [comment]
Given that conprof uses a noopAppender while starting we actually need to profile conprof startup with another conprof.
HA pair of conprof - anyone? 😂

**Comment by brancz** [comment]
We should be ok on Polar Signals to try this out as the sampler is separate from the storage (part of which would be restarting).

File: all.go [hunk 1]
```diff
@@ -97,69 +101,112 @@ func runAll(
...
+			)
+			return err
+		}()
+		if err != nil {
+			level.Error(logger).Log("msg", "failed to start conprof components", "err", err)
+			os.Exit(2)
+		}
+	}()
 
 	p.Ready()
```
Comment (brancz) [inline all.go:177]: We should only call ready when the DB is actually open.

Comment (metalmatze) [reply all.go:177]: Updated! :+1:

**Comment by brancz** [comment]
Generally looking good, but it looks like e2e tests are failing.

**Comment by metalmatze** [comment]
The e2e test are failing due to start up of HTTP and TSDB now being concurrent...
I need to take a look again to make it somewhat predictable for the tests.

**Pull request closed without merging**